uint32 id
string name
---
bool success
string message 